### PR TITLE
lets catslugs use spears

### PR DESCRIPTION
**Pull request by ynot01**
# Document the changes in your pull request

sentient catslugs only

they can throw them as god intended or use them in melee

# Changelog

:cl:  
tweak: catslugs can now wield spears 
/:cl:

### CODE REVIEW

**Review by cuackles — APPROVED** [review]
As God intended

File: code/modules/mob/living/simple_animal/friendly/catslug.dm [hunk 1]
```diff
@@ -16,3 +16,60 @@
...
+		weapon.forceMove(src)
+		melee_damage_lower = weapon.force + weapon.force_wielded
+		melee_damage_upper = weapon.force + weapon.force_wielded
+		armour_penetration = weapon.armour_penetration
+		melee_damage_type = weapon.damtype
+		sharpness = weapon.sharpness
+		attack_sound = weapon.hitsound
+		update_icons()
+	else if(!weapon && !istype(A, /obj/item/twohanded/spear))
+		to_chat(src, span_warning("You do not know how to wield the [A]!"))
```
Comment (tattax) [inline code/modules/mob/living/simple_animal/friendly/catslug.dm:44]:
```suggestion
	if(weapon)
		return
	if(istype(A, /obj/item/twohanded/spear))
		visible_message(span_notice("[src] wields the [A]."), span_notice("You wield the [A]."))
		weapon = A
		weapon.forceMove(src)
		melee_damage_lower = weapon.force + weapon.force_wielded
		melee_damage_upper = weapon.force + weapon.force_wielded
		armour_penetration = weapon.armour_penetration
		melee_damage_type = weapon.damtype
		sharpness = weapon.sharpness
		attack_sound = weapon.hitsound
		update_icons()
	else
		to_chat(src, span_warning("You do not know how to wield the [A]!"))
```

Comment (tattax) [reply code/modules/mob/living/simple_animal/friendly/catslug.dm:44]: do the messages look fine? we don't want the the here

File: code/modules/mob/living/simple_animal/friendly/catslug.dm [hunk 2]
```diff
@@ -16,3 +16,60 @@
...
+		update_icons()
+
+/mob/living/simple_animal/pet/catslug/update_icons()
+	. = ..()
+	if(stat == DEAD || resting)
+		return
+	if(weapon)
+		icon_state = "catslug_spear"
+	else
+		icon_state = "catslug"
```
Comment (tattax) [inline code/modules/mob/living/simple_animal/friendly/catslug.dm]:
```suggestion
	icon_state = weapon ? initial(icon_state) + "_spear" : initial(icon_state)
```

File: code/modules/mob/living/simple_animal/friendly/catslug.dm [hunk 3]
```diff
@@ -16,3 +16,60 @@
...
+
+/mob/living/simple_animal/pet/catslug/death(gibbed)
+	. = ..()
+	if(weapon)
+		weapon.forceMove(get_turf(src))
+		weapon = null
+	update_icons()
```
Comment (tattax) [inline code/modules/mob/living/simple_animal/friendly/catslug.dm]: might want to call parent later

File: code/modules/mob/living/simple_animal/friendly/catslug.dm [hunk 4]
```diff
@@ -16,3 +16,60 @@
...
+		melee_damage_lower = initial(melee_damage_lower)
+		melee_damage_upper = initial(melee_damage_upper)
+		armour_penetration = initial(armour_penetration)
+		melee_damage_type = initial(melee_damage_type)
+		sharpness = initial(sharpness)
+		attack_sound = initial(attack_sound)
+		weapon.forceMove(get_turf(src))
+		weapon.throw_at(A, weapon.throw_range, weapon.throw_speed, src)
+		weapon = null
+		update_icons()
```
Comment (tattax) [inline code/modules/mob/living/simple_animal/friendly/catslug.dm]:
```suggestion
	if(!weapon)
		return
	visible_message(span_warning("[src] throws the [weapon] at [A]!"), span_warning("You throw the [weapon] at [A]!"))
	melee_damage_lower = initial(melee_damage_lower)
	melee_damage_upper = initial(melee_damage_upper)
	armour_penetration = initial(armour_penetration)
	melee_damage_type = initial(melee_damage_type)
	sharpness = initial(sharpness)
	attack_sound = initial(attack_sound)
	weapon.forceMove(get_turf(src))
	weapon.throw_at(A, weapon.throw_range, weapon.throw_speed, src)
	weapon = null
	update_icons()
```

File: code/modules/mob/living/simple_animal/friendly/catslug.dm (hunk 1 above)
Comment (ynot01) [reply code/modules/mob/living/simple_animal/friendly/catslug.dm:44]: yes

File: code/modules/mob/living/simple_animal/friendly/catslug.dm (hunk 3 above)
Comment (ynot01) [reply code/modules/mob/living/simple_animal/friendly/catslug.dm]: i dont see why

**Review by Altoids1 — APPROVED** [review]
As Allah intended

**Comment by JamieD1** [comment]
What the fuck is a catslug and why is it in my spaceman game

**Pull request merged**